src/backend/backend.c

```diff
@@ -55,6 +60,59 @@ region_t get_damage(session_t *ps, bool all_damage) {
 	return region;
 }
 
+void get_resized_damage(session_t *ps, int stacking_rank, bool whole_screen,
+                        region_t *reg_damage, region_t *reg_paint) {
+	*reg_damage = get_damage(ps, whole_screen);
+
+	if (whole_screen) {
+		pixman_region32_init(reg_paint);
+		pixman_region32_copy(reg_paint, reg_damage);
+		return;
+	}
+
+	if (!pixman_region32_not_empty(reg_damage)) {
+		// Empty means there is nothing to paint, so we don't need to resize
+		// damage.
+		pixman_region32_init(reg_paint);
+		return;
+	}
+
+	/// The adjusted damaged regions
+	assert(ps->o.blur_method != BLUR_METHOD_INVALID);
+	if (ps->o.blur_method != BLUR_METHOD_NONE && ps->backend_data->ops->get_blur_size) {
+		int blur_width, blur_height;
+		ps->backend_data->ops->get_blur_size(ps->backend_blur_context,
+		                                     &blur_width, &blur_height);
+
+		// The region of screen a given window influences will be smeared
+		// out by blur. With more windows on top of the given window, the
+		// influences region will be smeared out more.
+		//
+		// Also, blurring requires data slightly outside the area that needs
+		// to be blurred. The more semi-transparent windows are stacked on top
+		// of each other, the larger the area will be.
+		//
+		// Instead of accurately calculate how much bigger the damage
+		// region will be because of blur, we assume the worst case here.
+		// That is, the damaged window is at the bottom of the stack, and
+		// all other windows have semi-transparent background
+		int resize_factor = 1;
+		if (stacking_rank) {
+			resize_factor = stacking_rank;
+		}
```

Comment:
`int resize_factor = stacking_rank || 1;`?

Reply:
No, this is more difficult to understand.

Reply:
Also this is not how C's `||` operator works

Reply:
oh, i see, my bad